torch_ttnn/passes/memory_pass.py

```diff
@@ -0,0 +1,240 @@
+
+    op_registry = OpRegistry()
+    mm.logs += "Track memory footprint for each of the ops:\n"
+    print("Track memory footprint for each of the ops:")
+    nodes = list(gm.graph.nodes)
+
+    # Tensor ID allocation tracker
+    last_assigned_tid = 1
+    # Memory address allocation tracker
+    last_assigned_addr = 0
+    # List of already executed nodes on device
+    computed_nodes = []
+    # Data points containing info of on device tensors allocation & deallocation for ttnn ops
+    # Useful for visualization by plotting in a chart
+    data_points = {}
+    # On device tensors info
+    on_device_meta = []
+
+    # Tensor IDs allocation for ttnn ops input & output
+    tt_compute_count = 0
+    for node in nodes:
+        if is_tt_compute(node):
+            tt_compute_count += 1
+            # For inputs
+            for input_node in node.all_input_nodes:
+                if input_node not in mm.node_to_tid_map:
+                    mm.node_to_tid_map[input_node] = last_assigned_tid
+                    last_assigned_tid += 1
+            # For output
+            if node not in mm.node_to_tid_map:
+                mm.node_to_tid_map[node] = last_assigned_tid
+                last_assigned_tid += 1
+
+    # print(f"Tensor IDs for nodes on device:")
+    # for key, value in mm.node_to_tid_map.items():
+    #     print(f"Node: {key} - {value}")
+
+    # Tracks the total compute memory of the model
+    mm.logs += f"Total tt compute nodes: {tt_compute_count}\n"
+    print(f"Total tt compute nodes: {tt_compute_count}")
+    i = 0
+    overflow_ops = []
+
+    for node in nodes:
+        on_device = False
+        # Has to be a ttnn op
+        if is_tt_compute(node):
+            mm.logs += "---------------------------------------------------\n"
+            print("\n---------------------------------------------------")
+            mm.logs += f"tt compute node numbered: {i}\n"
+            print(f"tt compute node numbered: {i}")
+            i += 1
+            # Exception: Full node has no input
+            # Assumption: Full node outputs a tensor on device
+            if node.target == ttnn.full:
+                on_device = True
+            total_input_size = 0
+            # Input & Output tensor IDs for ttnn op
+            ttnn_ops_tids = []
+            for input_node in node.all_input_nodes:
+                # If input tensor on device or coming as output from another ttnn on device op
+                if op_registry.is_input_tensor_on_device(input_node):
+                    on_device = True
+                    input_shape = op_registry.get_shape(input_node)
+                    input_dtype = op_registry.get_dtype(input_node)
+                    input_size = op_registry.get_size(input_shape, input_dtype)
+                    total_input_size += input_size
+
+                    assert (
+                        input_node in mm.node_to_tid_map
+                    ), "Tensor ID not allocated for one of the inputs!"
```

Comment:
This is not actionable. Can you maybe add details to the assert such that it will be more actionable?

Reply:
I am not sure what details to add in this assert. This assert simply checks if the tensor corresponding to input_node has a tensor ID or not. Generally speaking it should have because every ttnn op has its input & output tensor marked with a tensor ID. If it doesn't then raise an alarm.